### PR TITLE
PTL-1461: Changes to CSV Ingress How To... as a result of workshop

### PR DESCRIPTION
**Pull request by wjhendry**
Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?
Current

Have you checked all new or changed links?
Yes

THIS IS WHAT HAS CHANGED
- Information removed from **ingest-csv.mdx** so that we reach the reference to the example earlier
- key useful info from here has been dropped into **data-pipeline/introduction.mdx**. This will cause conflicts with https://github.com/genesiscommunitysuccess/docs/pull/1783 so I have kept the information in a separate .txt file
- added new **run-examples.mdx** with instructions (for now) for how to run locally. I have not been able to complete this exercise myself
- **embed.mdx** has been removed from the sidebar, as requested, but the file is still in the folder (as are several others)

### CODE REVIEW

File: docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx [hunk 1]
```diff
@@ -10,21 +10,90 @@ tags:
...
+Genesis Data Pipelines enable you to stream data into and out of your Genesis application. If you are looking to integrate your application with an external database or file system, this module should be your first consideration. If Data Pipeline does not offer you the solution you need, we suggest using [Apache Camel](/server/integration/apache-camel/introduction/) to create the integration. 
 
 Each Data Pipeline defines a source, some mapping operations and a sink.
 
 | term | description |
 | ---- | ----------- |
 | source | Changes in data at this location trigger your Data Pipeline |
-| map | Transforms data from the data source |
+| operators | Transform data from the data source |
 | sink | Does something with the mapped data from the data source |
```
Comment (gentaDemnushaj) [inline docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx:21]: I wasn't sure if this statement here needed to be more specific.:
>Does something with the mapped data from the data source

As I don't know much about it myself, best to double check here.

**Comment by wjhendry** [comment]
Yes, I know that statement! Well spotted.
I believe this will be changed in the new docs.
If it is still there tomorrow, I'll demand an explanation.
J




[cid:genesis-logo_61cae3c1-f120-4dce-aa4a-2cc1c66974cf.png]<https://genesis.global/?utm_campaign=FY2024_BRAND_Marketing&utm_source=employee_email_sig&utm_medium=email>

John Hendry | Senior Technical Writer
[cid:genesis8.3emailsignature(1)_bb0767c7-d5e8-42c4-8359-d519a35970b4.png]<https://genesis.global/blog/genesis-application-platform-version-8-3/?utm_campaign=FY2024_SEGMENT_Financial_Markets&utm_source=employee_email_sig&utm_medium=email>
<https://genesis.global/company/events/next-gen-mastering-tech-in-financial-markets-event/?utm_campaign=FY2024_EVENT_Genesis_NYC_USA_Aug&utm_source=employee_email_sig&utm_medium=email><https://genesis.global/?utm_campaign=FY2024_BRAND_Marketing&utm_source=employee_email_sig&utm_medium=email>
Follow us:      [cid:social-bug-linkedin_1004256a-c1de-4c1f-8a45-802f794510af.png] <https://www.linkedin.com/company/genesis-global-technology/>

Visit us: genesis.global<https://genesis.global/?utm_campaign=FY2024_BRAND_Marketing&utm_source=employee_email_sig&utm_medium=email>

From: Genta Demnushaj ***@***.***>
Sent: 26 September 2024 14:54
To: genesiscommunitysuccess/docs ***@***.***>
Cc: John Hendry ***@***.***>; Author ***@***.***>
Subject: [Ext] Re: [genesiscommunitysuccess/docs] PTL-1461: Changes to CSV Ingress How To... as a result of workshop (PR #1787)

External email. Use caution when clicking links or opening file attachments.


@gentaDemnushaj commented on this pull request.

________________________________

In docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx<https://github.com/genesiscommunitysuccess/docs/pull/1787#discussion_r1777126762>:

>

 Each Data Pipeline defines a source, some mapping operations and a sink.



 | term | description |

 | ---- | ----------- |

 | source | Changes in data at this location trigger your Data Pipeline |

-| map | Transforms data from the data source |

+| operators | Transform data from the data source |

 | sink | Does something with the mapped data from the data source |

I wasn't sure if this statement here needed to be more specific.:

Does something with the mapped data from the data source

As I don't know much about it myself, best to double check here.

-

File: docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx [hunk 2]
```diff
@@ -10,21 +10,90 @@ tags:
...
+                })
+            }
+```
+
+The example above has two operators. Let's look at them more closely.
+
+- `split` uses the pre-built decoder which converts the CSV file and outputs a map containing each row in column + value: `CsvRow<Map<String, String>>`
+- `map` takes the data we need from the row columns and formats it accordingly. We are building up ExchangeHoliday, a database entity in our app, wrapped in the Db operation we want to perform on it; in this case, we want to upsert.
+
+For more complex requirements, you can create your own [transformer](#keeping-the-data-transformation-separate) in a seperate file. You can see this in two of our examples.
```
Comment (gentaDemnushaj) [inline docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx:59]: the link here for 'transformer' doesn't take me anywhere else.

File: docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx [hunk 3]
```diff
@@ -10,21 +10,90 @@ tags:
...
+
+Also, you can build multiple database entities from a single file. You can see details of this in the [third example](#updating-multiple-different-entities-from-a-single-source).
+
```
Comment (gentaDemnushaj) [inline docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx:62]: Same thing with the link here. Are the examples being shown here?

File: docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx [hunk 4]
```diff
@@ -10,21 +10,90 @@ tags:
...
+The example sink below takes the output from `.map` (a set of DB operations) and handles the updating of the database.
+
+```kotlin
+.sink(dbSink())
+```
+
+That’s all you need to ingest the data and put it in your database.
+
+However, there are additional helpers for things such as `onCompletion`. This enables you specify what happens after the file processing is complete.
```
Comment (gentaDemnushaj) [inline docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx:74]: I think there's a typo here: 'Enables you to specify...'

Comment (wjhendry) [reply docs/03_server/11_integration/07_data-pipeline/01_introduction.mdx]: Thanks, extra word inserted

**Comment by perryking** [comment]
Closing - we'll write a new job to manage master

**Pull request closed without merging**